### ISSUE TITLE
Excel: Read item description text in Insert function (and subsequent function arguments) dialogs

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#6745]
In Excel, press SHIFT+F3 to open the "Insert function" dialog.  This dialog allows you to search for a function to meet a particular need.

Type, for instance, "Round up" and press enter to find functions which round numbers.  When you do this, the focus moves into the "Select a function" list.  As you arrow through this list, visually, there are two lines of text below the list, the first which shows the current function and how to format arguments.  The second is a simple description of what the function does.

This information is really useful when searching for a function, however it is fiddly to get to.  The easiest way currently is to press TAB once, then NVDA+numpad 4 (NVDA+shift+left arrow) to move to the previous object.  SHIFT+TAB can then be used to get back to the list.  This either needs to be done for each function, or NVDA+7 can be pressed to turn focus moves navigator object off when on the description (but you then need to remember to turn it back on, as you need it again on the next dialog!).

Once you find a function and press ENTER to activate the OK button.  A new dialog appears where you can add the arguments for the function. This basically works in the same way - there is a description for each argument field visually below the argument edit boxes.  When the dialog appears the focus is in the first argument edit.  Press NVDA+numpad 4 once to move to the name of the argument, and a second time to move to the description.  Again NVDA+7 can be used to hold the review cursor here.

It would be great if there was an easy (and consistent) way of reading this information.

**Comment by Adriani90** [comment]
I agree on this feature. However, for me it works also great with object navigation because I actually do not want to hear the descriptions every time when I press up and down arrow in the lists. However, it would be nice if there was a key stroke associated to the descriptions. But I guess this can only be done by Microsoft.

**Comment by Adriani90** [comment]
In Excel 2010 it works actually quite good. You select the function with arrow key and the description is automatically read by NVDA when you press tab once. Then press shift+tab, choose another function and press tab to hear its description.

**Comment by Qchristensen** [comment]
Actually, the first part of this now works in NVDA 2019.1.1 and Office 365 v16.0.11601.20174 .
If I press SHIFT+F3, type round up and press ENTER, the "Select a function" list changes to show matching functions, the focus moves to the first one (ceiling) and NVDA reads the description.  If I ARROW through the list, NVDA reads the name and description for each item.

The next window, where you can add the arguments for the function, still behaves as previously.  The name of each argument is read as the label to each edit box, so that is good.  Just to the right of each edit box, is another label with = and the current value based on what is typed in that edit.  For instance, if I type "B2", then the value of B2 will be printed to the right of the edit.

**Comment by Adriani90** [comment]
This seems to happen because the value of the cell entered is not part of the edit field, so there is no object hierarchy between edit field and the value displayed on the right side, they don't seem to be connected.

cc: @msftedad can you please forward this to your development team?

Steps to reproduce with NvDA 2025.1.2 and Office 365 version 2505:
1. Open excel
2. Fill some cells with data
3. press shift+f3
4. Choose a function (e.g. arithmetic average)
5. Press enter
6. In the arguments dialog, enter the coordinates of a cell which contains a value
7. Press tab and shift+tab to focus the edit field where you just entered the cell coordinates

actual: NVDA doesn't report the cell coordinates in the editfield, and it doesn't report the actual cell content at the coordinates.

Expected: NVDA should report something like "number 1 edit field; b3... <cell content>
In my case "number 1 edit field B3... 2
Because I entered 2 in cell b3 before creating the function.

**Comment by msftedad** [comment]
Hi @Adriani90, This issue seems to be specific to NVDA, as other screen readers are correctly reading the edit value information. The product team has confirmed that this issue is NVDA-specific.
